### ISSUE TITLE
[Feature Request]: Download code as zip button.

### ISSUE THREAD

**Issue by MeetDOD** [VaibhavArora314/StyleShare#196]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I want to add a button near the code snippet where when user click on it the code in the code snippet of that post will get downloaded in their machine.

Later they can do whatever they want with the zip.

### Use Case

User comfort

### Benefits

User flexibility

### Add ScreenShots

![image](https://github.com/VaibhavArora314/StyleShare/assets/120380073/66fe8bbb-9c2f-4f4f-8f9c-bdda0c46e484)

here i will give the Download as zip button.

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

**Comment by VaibhavArora314** [comment]
Currently, since there is only HTML code, I don't see much point in this.
This will be useful after #93, so let's wait for that. After that, you can implement this.

**Comment by MeetDOD** [comment]
@VaibhavArora314 I know but let consider a scenario when user what to host a tailwindcss website. In netlify you can just upoad your code folder and it will auto deploy it in that case this feature will be huge help even if that is small component :)

**Comment by MeetDOD** [comment]
@VaibhavArora314 Can you list down more features, I am not getting any better features to implement :(

**Comment by VaibhavArora314** [comment]
> @VaibhavArora314 Can you list down more features, I am not getting any better features to implement :(

![image](https://github.com/VaibhavArora314/StyleShare/assets/84830429/4acd1b40-afd1-47b3-ac41-5d8461bddd32)
navbar is not responsive due to so many options. maybe you can resolve this.

**Comment by MeetDOD** [comment]
@VaibhavArora314 should i raise new issue 🤔

**Comment by VaibhavArora314** [comment]
@MeetDOD once the js css options have been added in #93, then you can implement this.

**Comment by MeetDOD** [comment]
Okay then i am eagerly waiting :)

**Comment by MeetDOD** [comment]
@VaibhavArora314 should i implement this feature now, because js is introduce now in style share ?

**Comment by VaibhavArora314** [comment]
> @VaibhavArora314 should i implement this feature now, because js is introduce now in style share ?

Sure
